src/compiler/emitter.ts

```diff
@@ -2777,34 +2776,61 @@ var __param = this.__param || function(index, decorator) { return function (targ
                 return getCombinedNodeFlags(node.parent);
             }
 
-            function renameNonTopLevelLetAndConst(node: Node): void {
+            function renameBlockScopedValueIfNecessary(node: Node, meaning: SymbolFlags): void {

```

Comment:
I will rename it to `expectedMeaning` - the intent behind this parameter is to denote the expected meaning for the name defined by `node`